src/parser.rs

```diff
@@ -69,10 +69,40 @@ pub enum Header {
     Item(Item),
 }
 
+pub enum HType {
+    L,
+    D,
+    I,
+}
+
 pub struct Parser;
 
 impl Parser {
-    pub fn parse(input_bytes: &[u8], header_type: &str) -> ParserResult<Header> {
+    pub fn parse_dict_header(input_bytes: &[u8]) -> ParserResult<Dictionary> {
+        if let Header::Dictionary(dict) = Self::parse(input_bytes, HType::D)? {
+            Ok(dict)
+        } else {
+            Err("not a dictionary header")
+        }
+    }
+
+    pub fn parse_list_header(input_bytes: &[u8]) -> ParserResult<List> {
+        if let Header::List(dict) = Self::parse(input_bytes, HType::L)? {
```

Comment:
maybe not name them dict. 🙂

Reply:
Very good comment)) But I'm not happy with the whole change, to be honest